src/get_nwbfile_info/core.py

```diff
@@ -200,134 +212,102 @@ def process_nwb_container(obj, path="nwb", visited=None):
                 item_path = f"{path}[{i}]"
                 results.extend(process_nwb_container(item, item_path, visited))
         except Exception as e:
-            # Raise a warning if we can't iterate
             warnings.warn(f"Could not iterate through {path}: {e}")
 
     return results
 
-
-def process_dict_like(obj, path, visited):
+def get_nwbfile_usage_script(url_or_path):
     """
-    Process dictionary-like objects (including LabelledDict) and generate Python code to access their items.
-
-    Args:
-        obj: The dictionary-like object to process
-        path: The Python code path to access this object
-        visited: Set of already visited object IDs to avoid cycles
-
-    Returns:
-        List of strings with Python code to access the object's items
-    """
-    results = []
-
-    # Try to iterate through items
-    for key, value in obj.items():
-        # Skip private keys
-        if isinstance(key, str) and key.startswith('_'):
-            continue
-
-        # Format the path based on the key type
-        if isinstance(key, str):
-            item_path = f"{path}[\"{key}\"]"
-        else:
-            item_path = f"{path}[{key}]"
-
-        # AbstractContainer objects will be printed later in recursion
-        if not isinstance(value, hdmf.container.AbstractContainer):
-            type_name = get_type_name(value)
-            results.append(f"{item_path} # ({type_name}) L238")
-
-        # Recursively process the value
-        results.extend(process_nwb_container(value, item_path, visited))
-
-    return results
-
-
-def analyze_nwb_file(url):
+    Analyze an NWB file and return Python code to access its objects and fields.
     """
-    Analyze an NWB file and print Python code to access its objects and fields.
-
-    Args:
-        url: URL to the NWB file
+    is_url = url_or_path.startswith(('http://', 'https://'))
+    is_lindi = url_or_path.endswith('.lindi.json') or url_or_path.endswith('.lindi.tar')
```

Comment:
```suggestion
    is_lindi = url_or_path.endswith(('.lindi.json', '.lindi.tar'))
```
I didn't know about this syntax which you use in the line above. Convenient!